json_configs/configuration.h

```diff
@@ -43,123 +43,950 @@ namespace vega_config {
 		void load_config(const boost::filesystem::path & config_file_path);
 
 
+    struct {
+        std::string details = "High";
+        std::string shaders = "Average";
+        std::string audio = "All Sounds";
+        std::string physics = "Default (1.0)";
+
+    } settings_app;
+
     struct {
 
     } advanced;
 
+    struct {
+
+        struct {
+            std::string key = "S";
+            std::string modifier = "none";
+
+        } Cockpit__SkipMusicTrack;
```

Comment:
this would be better served using a named type here so you could do something like:
```
struct KeySet {
    std::string key;
    std::string modifier;
};

KeySet Cockpit__SkipMusicTrack{key("S"), modifier("none")};
KeySet Screenshot{key("i"), modifier("none")};
...
```

Not sure if I have the initializers right but something like that.

Reply:
Maybe. I'll have to think about how to implement that in JsonStruct though.